Lombiq.Tests.UI/Docs/Tools.md

```diff
@@ -15,3 +15,4 @@
 - Monkey testing is implemented using [Gremlins.js](https://github.com/marmelab/gremlins.js/) library.
 - Visual verification is implemented using [ImageSharpCompare](https://github.com/Codeuctivity/ImageSharp.Compare).
 - [Ben.Demystifier](https://github.com/benaadams/Ben.Demystifier) is used to simplify stack traces, mainly around async methods.
+- Creating element screenshot using [Noksa.WebDriver.ScreenshotsExtensions](https://github.com/Noksa/WebDriver.Screenshots.Extensions).
```

Comment:
Ok, then we should implement this feature, or we can fork and maintain it.

Reply:
Why do we really need it though, exactly?

Reply:
`Selenium` creates a screenshot only from the visible area, but we need a full-page screenshot for full-page validation.
I started to implement it, but meantime I found this lib which is working well. This is why I implemented `WaitScrollToNotChange` and some other scroll related functions.

Reply:
I see. Hmm, if there's no better way to do this then let's keep it.

Reply:
I see, then please do that.

Reply:
Finally, I've found two workarounds to make the full-page comparisons better. Both of them are related to scrollbars, and change the look and maybe the behavior of the page.

1. Calculating the scrollbar size, and using its value, when setting the viewport size. This does not look like the best choice, because of the automatic text alignment. When the window size changes, the rendered number of lines could be less, and finally, the scrollbars could get disappeared independently, and the width of the client area gets bigger than expected.
2. Hiding the scrollbars with `overflow: hidden`. This could affect the rendered number of lines too, but we have more control over the width of the client area.

So, I implemented the second workaround, you can see it in `VerifyHomePage`.

I was thinking about that, how can we use `AssertVisualVerificationApproved` or `AssertVisualVerification` to validate rendered content on every page change like `AssertHtmlValidityAsync`, and I found the following problems:
1. The `AfterPageChange` event handler will be the first method in the call stack that is not marked with `VisualVerificationApprovedMethodAttribute`. We can filter out the event handler method and maybe the methods between the event handler and `GoToAsync`, but where is the limit? - a new attribute can help us to mark the test method explicitly.
2. We can use the absolute Uri as a suffix for the reference images, but what if the test navigates to the same Uri more than once? - a counter or hash maybe help, I'm not sure.

Reply:
The approach in `VerifyHomePage` looks good!

Let's not complicate this any further. Previously you've said that we'd need a high error tolerance for full-page tests; I assume that's still the case, even if slightly lower, with the scrollbar hidden? If indeed, then we don't need automatic full-page tests yet.

Reply:
I'm not sure I'm following: Did you solve the large error tolerance with removing the scrollbar?

Reply:
That seems like quite a good solution to me :). Let's address this separately later though, when we have more experience: https://github.com/Lombiq/UI-Testing-Toolbox/issues/179